runtime/lib/backend/openqasm/OpenQasmDevice.cpp

```diff
@@ -473,9 +473,10 @@ void OpenQasmDevice::PartialCounts(DataView<double, 1> &eigvals, DataView<int64_
     }
 }
 
-auto OpenQasmDevice::Measure([[maybe_unused]] QubitIdType wire, int8_t postselect) -> Result
+auto OpenQasmDevice::Measure([[maybe_unused]] QubitIdType wire, std::optional<int8_t> postselect)
+    -> Result
 {
-    RT_FAIL_IF(postselect != -1, "Post-selection is not supported yet");
+    RT_FAIL_IF(postselect && *postselect != -1, "Post-selection is not supported yet");
```

Comment:
Same here

Reply:
Recall we are sending a -1 here: https://github.com/PennyLaneAI/catalyst/blob/0509f9ebb71078f32d3bb9a4250c05943a6f868c/mlir/lib/Quantum/Transforms/ConversionPatterns.cpp#L539

Reply:
This is where the C-API is useful. It can convert between the C pattern (-1) to the C++ pattern (Optional). Then all downstream C++ code does not need to deal with the C convention.

Reply:
Got it. WIP :)

Reply:
Done!

Reply:
Nice 🚀